app/controllers/internal/events_controller.rb

```diff
@@ -42,12 +44,38 @@ def approve
       GetIntoTeachingApiClient::TeachingEventsApi.new.upsert_teaching_event(api_event)
       Rails.logger.info("#{@user.username} - publish - #{api_event.to_json}")
       redirect_to internal_events_path(
-        success: :open,
+        success: :published,
         event_type: determine_event_type_from_id(api_event.type_id),
         readable_id: api_event.readable_id,
       )
     end
 
+    def withdraw
+      api_event = GetIntoTeachingApiClient::TeachingEventsApi.new.get_teaching_event(params[:id])
+      api_event.status_id = GetIntoTeachingApiClient::Constants::EVENT_STATUS["Pending"]
+      GetIntoTeachingApiClient::TeachingEventsApi.new.upsert_teaching_event(api_event)
+      Rails.logger.info("#{@user.username} - withdrawn - #{api_event.to_json}")
+      redirect_to internal_events_path(
+        success: :withdrawn,
+        event_type: determine_event_type_from_id(api_event.type_id),
+        readable_id: api_event.readable_id,
+      )
+    end
+
+    def live_events
+      events = GetIntoTeachingApiClient::TeachingEventsApi
+                 .new
+                 .search_teaching_events_grouped_by_type(quantity_per_type: 1_000,
+                                                         start_after: DateTime.now.utc.beginning_of_day,
+                                                         status_ids: [open_event_status_id])
+      events = events.select do |key|
+        key.type_id == event_types[:provider] || key.type_id == event_types[:online]
+      end
+      if events[0] && events[1]
```

Comment:
What if there are only live events of one type? Could you do something like:

```
@live_events = events.map(&:teaching_events).flatten
```

Does it make sense to use the CRM notion of 'open' instead of 'live' for consistency?